sdks/python/apache_beam/testing/analyzers/perf_analysis.py

```diff
@@ -84,6 +85,20 @@ def run_change_point_analysis(params, test_name, big_query_metrics_fetcher):
   if not change_point_index:
     logging.info("Change point is not detected for the test %s" % test_name)
     return False
+  # Remove the change points that are at the edges of the data.
+  # https://github.com/apache/beam/issues/28757
+  # Remove this workaround once we have a good solution to deal
+  # with the edge change points.
+  if is_edge_change_point(change_point_index,
```

Comment:
I would move this logic into find_latest_change_point_index since there are other considerations in that function to filter out noise.

Reply:
you could also skip adding extra param to find_latest_change_point_index until we have a usecase to customize it with a non-default value.

Reply:
Done